### PR TITLE
fix(anac-pl): accept the CPV check digit in the advanced search

### PR DESCRIPTION
**Pull request by aborruso**
ANAC's release of 15 September made the advanced search accept a CPV code in the form it has in official documents, with the check digit (`30213000-5`). The CLI still refused it before calling the service.

- **`cerca-avanzata --cpv` and `affidamenti --cpv-code`** now accept the check digit on a full 8-digit code. Measured on the service (publications 01/06-12/09/2026): `30213000`, `30213000-5` and `30213000-9` all return the same 31 notices, so the check digit is ignored, right or wrong.
- **On a prefix** the hyphen returns nothing without explanation (`302-5` gives zero results), so the CLI keeps refusing it with a usage error that says why.
- Tests for the edge cases; flag help, README and SKILL updated; the existing `anac-pl-cerca-avanzata-cpv` patch record extended.
- README: install from the catalog written in the present tense, with the `update` command.

Checks: `go build`, `go vet`, `go test ./...` and govulncheck in the package; live gate `dogfood --live --level full` 66/66, 0 failed; acceptance proof hashes match the packaged files (90/90, module path mapped back).

### CODE REVIEW

File: library/other/anac-pl/internal/cli/affidamenti.go [hunk 1]
```diff
@@ -354,7 +354,7 @@ affidamenti aggregati via Consip; incrociare con altre fonti (es. MxMap).
 	f.StringVarP(&query, "query", "q", "", "Testo libero (keyword, CIG, CUP, oggetto)")
 	f.StringVarP(&tipologia, "tipologia", "t", "esiti", "Tipologia (nome/slug o template); default 'esiti'")
 	f.StringVar(&cpv, "cpv", "", "Valore grezzo per il campo CPV della vecchia ricerca (match testuale, non selettivo; di norma usa --cpv-code)")
-	f.StringVar(&cpvCode, "cpv-code", "", "Codice CPV o suo prefisso (min 3 cifre): usa la ricerca avanzata ANAC, che filtra davvero per codice. Non combinabile con --query/--amount-*")
+	f.StringVar(&cpvCode, "cpv-code", "", "Codice CPV o suo prefisso (min 3 cifre), anche con la cifra di controllo (30213000-5): usa la ricerca avanzata ANAC, che filtra davvero per codice. Non combinabile con --query/--amount-*")
```
[the remark this answers is not in the document]
Comment (aborruso) [reply library/other/anac-pl/internal/cli/affidamenti.go:376]: Right, fixed in f873be6, and it was hiding a larger problem.

The filter compared the whole flag value, so `30213000-5`, and also a list such as `72412000,30213000`, could never match a row CPV. It now compares each code of the list with the check digit stripped.

It did not show up in practice because the row CPV was empty on every row: the notice detail now exposes the CPV as `72412000_Fornitori di servizi di posta elettronica`, which `NormalizeCPV` did not recognise, and empty CPVs are kept by the automatic filter. `NormalizeCPV` now parses that form too. Measured live with `affidamenti --cpv-code 72412000 -t "" --pages 2`: 38 rows, all with an empty `cpv` before, all `72412000` after; `72412000-9` gives the same 38 rows, `72412000,30213000` gives 29 (28 + 1). Tests cover the new form, the check digit and the list. Proof regenerated, 66/66.

File: library/other/anac-pl/internal/cli/cerca_avanzata.go [hunk 2]
```diff
@@ -248,13 +249,29 @@ func validateCPVFilter(cpv string) error {
...
+		if i := strings.IndexByte(t, '-'); i >= 0 {
+			base = t[:i]
+			check := t[i+1:]
+			if len(check) != 1 || check[0] < '0' || check[0] > '9' {
+				return fmt.Errorf("--cpv: dopo il trattino serve la sola cifra di controllo (es. 30213000-5), non %q", t)
+			}
+			if len(base) != 8 {
+				return fmt.Errorf("--cpv: la cifra di controllo vale solo col codice completo a 8 cifre (es. 30213000-5); con un prefisso come %q il servizio non restituisce nulla", t)
+			}
+		}
```
[the remark this answers is not in the document]
Comment (aborruso) [reply library/other/anac-pl/internal/cli/cerca_avanzata.go:269]: Right, fixed in f873be6. `soloCodiciCompleti` now goes through the same helper as the row filter, which splits on commas and strips the check digit, so `30213000-5` counts as a full 8-digit code and the underestimated-count warning no longer fires on it. Covered by a test alongside `30213000-5,42120000` and mixed lists with a prefix.

File: library/other/anac-pl/internal/cli/affidamenti.go [hunk 3]
```diff
@@ -322,10 +322,20 @@ affidamenti aggregati via Consip; incrociare con altre fonti (es. MxMap).
 			// che l'API ha correttamente restituito. Se invece --cpv-exact è
 			// esplicito, l'utente vuole un filtro stretto.
 			if cpvExact != "" {
-				want := strings.TrimSpace(cpvExact)
+				// Ogni codice della lista, senza cifra di controllo: con
+				// --cpv-code 30213000-5 o 30213000,42120000 il confronto sul
+				// valore intero non trovava mai i CPV normalizzati delle righe.
+				want := codiciCPV(cpvExact)
```
[the remark this answers is not in the document]
Comment (aborruso) [reply library/other/anac-pl/internal/cli/affidamenti.go:337]: Right, fixed in 35474ff80.

`--cpv-exact` now goes through `validateCPVFilter` before anything else, so `302-5` is refused with a usage error (exit 2) instead of being reduced to `302`; `72412000-9` is still accepted and keeps the one matching row. The validator's errors now name the flag actually used (`--cpv`, `--cpv-code`, `--cpv-exact`).

Since the same check-digit assumption could hide elsewhere, I went through every path that reads or compares a CPV and fixed the ones that were off: `cpv search 30213000-5` returned `null` and now finds the code; `NormalizeCPV` strips the check digit in the object form too; the `cpv` column of `cerca-avanzata` shows the normalised code rather than the raw value; `cpv get` on an unknown code exits 3 (not found) instead of 4 (auth). `cpv get 30213000-5`, `cerca --cpv` and `avvisi search --cpv` were already fine. Tests for each; proof regenerated, 66/66.

File: library/other/anac-pl/internal/cpvdata/cpvdata.go [hunk 4]
```diff
@@ -133,6 +152,11 @@ func Search(query string, limit int) []Entry {
 	if query == "" {
 		return nil
 	}
+	// Un codice con la cifra di controllo (30213000-5) è ancora una ricerca
+	// per codice: senza questo passaggio finiva fra le parole e non trovava nulla.
+	if i := strings.IndexByte(query, '-'); i > 0 && allDigits(query[:i]) && allDigits(query[i+1:]) {
+		query = query[:i]
+	}
```
[the remark this answers is not in the document]
Comment (aborruso) [reply library/other/anac-pl/internal/cpvdata/cpvdata.go]: Right, fixed in bf2c214.

`cpv search` and `cpv get` now recognise the check digit only in the exact form of 8 digits, hyphen, 1 digit (`CodiceConControllo`), the same rule the flag validator applies. Measured: `cpv search 30213000-55` and `302-55` now return 0 entries instead of 1 and 50; `cpv get 30213000-55` and `30213000-abc` exit 3 (not found), since `cpv get` had the same loose truncation, including non-numeric suffixes. `30213000-5` still resolves to 30213000. Tests cover the valid form and each malformed variant, and check that a malformed query never widens to the 302 family.

While checking the vocabulary against the CPV list ANAC publishes (anticorruzione/npa, `tipologiche/CPV.json`), I also found that `03116200` carried the description of `03117140`, which was missing. Fixed and confirmed on CPV 2008; the 9454 codes now match entry by entry. Proof regenerated, 66/66.
